torch/nn/parallel/scatter_gather.py

```diff
@@ -1,25 +1,36 @@
+import torch
 from torch.autograd import Variable
 from ._functions import Scatter, Gather
+from torch.cuda.comm import broadcast
 
 
-def scatter(input, target_gpus):
-    """Slices a given variable into approximately equal chunks and distributes
-       them accross given GPUs
+def scatter(input, target_gpus, dim=0):
+    """
+    Slices a given variable into approximately equal chunks and distributes
+      them accross given GPUs
+    Duplicates references to objects that are not variables.
     """
     def scatter_map(obj):
         if isinstance(obj, Variable):
-            return Scatter(target_gpus)(obj)
-        return tuple(zip(*map(scatter_map, obj)))
+            return Scatter(target_gpus, dim=dim)(obj)
+        if isinstance(obj, tuple) or isinstance(obj, list):
+            return type(obj)(zip(*map(scatter_map, obj)))
+        if torch.is_tensor(obj):
+            return broadcast(obj, target_gpus)
```

Comment:
Variables seem to be the only obvious thing to scatter. Also there's no obvious way to broadcast anything that would be contained in a tensor in data parallel. Not sure of a specific use case but could imagine an easy method to broadcast something instead of scattering could be useful.